### PR TITLE
Add user controller and support for show and update

### PR DESCRIPTION
**Pull request by enzofab91**
**Trello reference:**
https://trello.com/c/6ltAUJh0

**Description:**
This PR add all files required for get and modify user's attributes.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -56,6 +56,65 @@ FORMAT: 1A
...
+
+## Get User [GET]
+
++ Request Bad
+
+    + Headers
+
+            Content-Type: application/json
+
+    + Body
```
Comment (sebastiancaraballo) [inline apiary.apib:69]: 👀

File: apiary.apib [hunk 2]
```diff
@@ -56,6 +56,65 @@ FORMAT: 1A
...
+
+## Update User [PATCH]
+
++ Request Bad
+
+    + Headers
+
+            Content-Type: application/json
+
+    + Body
```
Comment (sebastiancaraballo) [inline apiary.apib:96]: 👀

File: app/views/api/v1/users/show.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,3 @@
+json.user do
+  json.partial! 'api/v1/users/info', user: current_user
+end
```
Comment (sebastiancaraballo) [inline app/views/api/v1/users/show.json.jbuilder:3]: `json.user current_user, partial: 'api/v1/users/info', as: :user`

File: db/migrate/20210428144426_add_target_count_to_users.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddTargetCountToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column('users', 'targets_count', :integer)
```
Comment (sebastiancaraballo) [inline db/migrate/20210428144426_add_target_count_to_users.rb:3]: Nice! However, it should have a default value and not allowed to be null.

File: spec/requests/api/v1/users/update_spec.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    expect(user.reload.last_name).to eq(params[:user][:last_name])
+  end
+
+  it 'returns the user' do
+    subject
+
+    expect(json[:user][:id]).to eq user.id
+    expect(json[:user][:first_name]).to eq user.first_name
+    expect(json[:user][:last_name]).to eq user.last_name
+    expect(json[:user][:gender]).to eq user.gender
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/users/update_spec.rb:33]: Ideally, each example should contain only one `expect`.

File: app/views/api/v1/users/_info.json.jbuilder [hunk 6]
```diff
@@ -0,0 +1,5 @@
+json.id           user.id
+json.email        user.email
+json.first_name   user.first_name
+json.last_name    user.last_name
+json.gender       user.gender
```
Comment (sebastiancaraballo) [inline app/views/api/v1/users/_info.json.jbuilder:5]: Here it's possible to use `json.extract!`

**Review by mrodriguez90 — APPROVED** [review]
👍

**Review by sebastiancaraballo — APPROVED** [review]

**Pull request merged**